platform/admin/admin-configuration-passwordencryption/src/test/java/org/codice/ddf/admin/configuration/password/encryption/PasswordEncryptingPersistenceManagerTest.java

```diff
@@ -0,0 +1,173 @@
+        when(configurationAdmin.listServices(anyString(),
+                anyString())).thenReturn(Collections.singletonList(metatype));
+    }
+
+    @Test
+    public void testLoad() throws IOException {
+        Dictionary dictionary = passwordEncryptingPersistenceManager.load("123");
+        assertThat(dictionary.get("password"), is("decrypted"));
+        assertThat(dictionary.get("notpassword"), is("foo"));
+    }
+
+    @Test
+    public void testLoadNoDecrypt() throws IOException {
+        Dictionary dictionary = passwordEncryptingPersistenceManager.load("321");
+        assertThat(dictionary.get("random"), is("stuff"));
+        verifyZeroInteractions(encryptionService);
+    }
+
+    @Test
+    public void testLoadNoDecryptNotString() throws IOException {
+        passwordEncryptingPersistenceManager.load("456");
+        verifyZeroInteractions(encryptionService);
+    }
+
+    @Test
+    public void testStore() throws IOException {
+        Hashtable decrypted = new Hashtable();
+        decrypted.put("password", "decrypted");
+        passwordEncryptingPersistenceManager.store("123", decrypted);
+        verify(encryptionService, times(1)).encrypt("decrypted");
+    }
+
+    @Test
+    public void testStoreMsf() throws IOException {
+        Hashtable decrypted = new Hashtable();
+        decrypted.put("password", "decrypted");
+        passwordEncryptingPersistenceManager.store("123.uuid", decrypted);
+        verify(encryptionService, times(1)).encrypt("decrypted");
+    }
+
+    @Test
+    public void testStoreNoService() throws IOException {
+        when(configurationAdmin.listServices(anyString(),
+                anyString())).thenReturn(new ArrayList<>());
+        Hashtable decrypted = new Hashtable();
+        decrypted.put("password", "decrypted");
+        passwordEncryptingPersistenceManager.store("123", decrypted);
+        verifyZeroInteractions(encryptionService);
+    }
+
+    @Test
+    public void testStoreNullService() throws IOException {
+        when(configurationAdmin.listServices(anyString(), anyString())).thenReturn(null);
+        Hashtable decrypted = new Hashtable();
+        decrypted.put("password", "decrypted");
+        passwordEncryptingPersistenceManager.store("123", decrypted);
+        verifyZeroInteractions(encryptionService);
+    }
+
+    @Test
+    public void testGetDictionaries() throws IOException {
+        passwordEncryptingPersistenceManager.getDictionaries();
+        verify(persistenceManager, times(1)).getDictionaries();
+    }
+
+    @Test
+    public void testExists() {
+        passwordEncryptingPersistenceManager.exists("123");
+        verify(persistenceManager, times(1)).exists("123");
+    }
+
+    @Test
+    public void testDelete() throws IOException {
+        passwordEncryptingPersistenceManager.delete("123");
+        verify(persistenceManager, times(1)).delete("123");
+    }
+
+    class ServiceImpl extends HashMap<String, Object> implements Service {
```

Comment:
I disagree, the test is only testing at the interface level and doesn't need to know anything about the actual impls, which is exactly why I can create my own impls here.